mlflow/store/artifact/azure_blob_artifact_repo.py

```
@@ -197,7 +197,23 @@ def _download_file(self, remote_file_path, local_path):
             blob.readinto(file)
 
     def delete_artifacts(self, artifact_path=None):
-        raise MlflowException("Not implemented yet")
+        from azure.core.exceptions import ResourceNotFoundError
+
+        (container, _, dest_path, _) = self.parse_wasbs_uri(self.artifact_uri)
+        container_client = self.client.get_container_client(container)
+        if artifact_path:
+            dest_path = posixpath.join(dest_path, artifact_path)
+
+        try:
+            blobs = container_client.list_blobs(name_starts_with=dest_path)
+            blob_list = list(blobs)
+            if not blob_list:
+                raise MlflowException(f"No such file or directory: '{dest_path}'")
```

Do we need to raise an exception here?

---

Never mind. `list_blobs` returns an empty iterator.